docs/gcloud-auth.rst

```diff
@@ -0,0 +1,264 @@
+In that case, jump up to the section about
+using a specific keyfile.
+Thanks to the order of evaluation,
+the keyfile you specify explicitly or via the environment variable
+will take precedence over the automatically discovered key in the
+App Engine or Compute Engine environment.
+
+=============================================
+Using multiple credentials in the same script
+=============================================
+
+There may be times where you want to use
+multiple different sets of credentials inside the same script.
+To do this, you should create miltiple connections
+and specify which to use on the various API calls you make.
+
+For example, here is how you would create two Pub/Sub topics
+in two different projects
+with two different sets of credentials.
+
+.. code-block:: python
+
+   from gcloud import pubsub
+   from gcloud.credentials import get_for_service_account_json
+
+   # Create two different credentials.
+   credentials1 = get_for_service_account_json('key1.json')
+   credentials2 = get_for_service_account_json('key2.json')
+
+   # Create two different connections.
+   connection1 = pubsub.Connection(credentials=credentials1)
+   connection2 = pubsub.Connection(credentials=credentials2)
+
+   # Create two different topics.
+   pubsub.Topic('topic1', project='project1', connection=connection1).create()
+   pubsub.Topic('topic2', project='project2', connection=connection2).create()
+
+If you have one "main" set of credentials
+and others are "one-offs" for special cases,
+you can avoid all this typing
+by using the default connection for the main set,
+and separate connections for the others.
+
+Using the same example as before,
+and assuming ``'key1.json'`` is the "main" set of credentials,
+here's what that same code might look like.
+
+.. code-block:: python
+
+   from gcloud import pubsub
+   from gcloud.credentials import get_for_service_account_json
+
+   credentials2 = get_for_service_account_json('key2.json')
+   connection2 = pubsub.Connection(credentials=credentials2)
+
+   pubsub.Topic('topic1').create()
+   pubsub.Topic('topic2', project='project2', connection=connection2).create()
+
+===============
+Troubleshooting
+===============
+
+--------------------------------------------------
+You keep mentioning key.json... How do I get that?
+--------------------------------------------------
+
+When we say ``key.json``
+we're referring to what we call a "Service Account".
+These are sort of like passwords
+which let you access the different cloud services.
+
+Service Accounts are pretty easy to create,
+just head over to the `Credentials`_ page
+in the `Google Developers Console`_,
+create a new Client ID,
+and generate a new JSON key.
+
+If you're a bit confused,
+take a look at the step by step guide here:
+https://cloud.google.com/storage/docs/authentication#service_accounts
+
+----------------------------------------------------------------
+I'm running in Compute Engine, but can't access Datastore. Help?
```

Comment:
We should make this more generic. We can include a stacktrace with a 403 error and just talk about turning services on in a project (getting a VM custom scopes is still a separate issue).